py/visdom/server.py

```diff
@@ -428,6 +432,142 @@ def on_close(self):
             self.subs.pop(self.sid, None)
 
 
+# TODO condense some of the functionality between this class and the
+# original SocketHandler class
```

Comment:
Agreed that it would be nice to reuse functionality between both implementations, but that can be done later